libs/vkd3d/command.c

```diff
@@ -6299,6 +6299,67 @@ static size_t get_query_stride(D3D12_QUERY_TYPE type)
     return sizeof(uint64_t);
 }
 
+static void d3d12_command_list_resolve_binary_occlusion_queries(struct d3d12_command_list *list,
+        struct d3d12_resource *buffer, UINT64 offset, UINT query_count)
+{
+    const struct vkd3d_query_ops *query_ops = &list->device->meta_ops.query;
+    const struct vkd3d_vk_device_procs *vk_procs = &list->device->vk_procs;
+    VkDescriptorBufferInfo buffer_info;
+    struct vkd3d_query_op_args args;
+    VkWriteDescriptorSet vk_write;
+    unsigned int num_workgroups;
+    VkMemoryBarrier vk_barrier;
+    VkDescriptorSet vk_set;
+
+    d3d12_command_list_invalidate_current_pipeline(list, true);
+    d3d12_command_list_invalidate_root_parameters(list, VK_PIPELINE_BIND_POINT_COMPUTE, true);
+
+    vk_set = d3d12_command_allocator_allocate_descriptor_set(list->allocator,
+            query_ops->vk_set_layout, VKD3D_DESCRIPTOR_POOL_TYPE_STATIC);
+
+    buffer_info.buffer = buffer->vk_buffer;
+    buffer_info.offset = buffer->heap_offset + offset;
```

Comment:
Is alignment a concern here?

Reply:
Right; we can only guarantee 8 bytes here. Guess we need to pass in an offset via push constants as well.

Reply:
Fixed.